Don't record QEMU's state in xenstore

When QEMU is running deprivileged, it does not have access to xenstore,
so don't attempt to use it to record the device model state.  Remove it
and have the toolstack talk to QEMU over QMP instead.

diff --git a/hw/xen/xen-common.c b/hw/xen/xen-common.c
index 4056420..cf58534 100644
--- a/hw/xen/xen-common.c
+++ b/hw/xen/xen-common.c
@@ -90,33 +90,6 @@ void xenstore_store_pv_console_info(int i, Chardev *chr)
     }
 }
 
-
-static void xenstore_record_dm_state(struct xs_handle *xs, const char *state)
-{
-    char path[50];
-
-    if (xs == NULL) {
-        fprintf(stderr, "xenstore connection not initialized\n");
-        exit(1);
-    }
-
-    snprintf(path, sizeof (path), "device-model/%u/state", xen_domid);
-    if (!xs_write(xs, XBT_NULL, path, state, strlen(state))) {
-        fprintf(stderr, "error recording dm state\n");
-        exit(1);
-    }
-}
-
-
-static void xen_change_state_handler(void *opaque, int running,
-                                     RunState state)
-{
-    if (running) {
-        /* record state running */
-        xenstore_record_dm_state(xenstore, "running");
-    }
-}
-
 void xen_setup_post(void)
 {
     int rc;
@@ -150,7 +123,6 @@ static int xen_init(MachineState *ms)
         xc_interface_close(xen_xc);
         return -1;
     }
-    qemu_add_vm_change_state_handler(xen_change_state_handler, NULL);
     return 0;
 }
 
